### PR TITLE
Ease widgets installation during web installation process

### PR DESCRIPTION
**Pull request by adr-mo**
## Description

For a few days now, Centreon has dependencies on the widgets.
It could be great to give the opportunity to the user to install the widgets (web part) at the end of the web installation (already done for centreon-license-manager, centreon-pp-manager, centreon-autodiscovery-server ...)

This PR aims also to standardize the style (checkboxes) with the latest version of centreon

**Fixes** None

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [x] 19.10.x (master)

<h2> How this pull request can be tested ? </h2>

When I get for the first time to the last step of the web installation
All the checkboxes should be checked.

**Use case 1**

- I directly click on “Install” button, which will trigger the installation of all modules and widgets
- The “Install” button should turn grey and the text changes from “Install” to “Installing…
- Once installation done, the button should be replaced by the “Next” button and all checkboxes are green.
- If I hit the “Refresh” button or F5, all the checkboxes should be green and the “Next” button should be there too.

**Use case 2**

- I uncheck some of the modules/widgets. The install button should be kept
- I click on the “Install” button
- The “Install” button should turn grey and the text changes from “Install” to “Installing…
- Once installation done, the button should be replaced by the “Next” button and all selected checkboxes should be green.

**Two choices here:**

- I click on Next to finalize the installation
- I click on other checkboxes to install other modules / widgets, then the install should appear again and so on…

**Use case 3**

- I uncheck all the modules/widgets
- The “Next” button should appear

#### Community contributors & Centreon team

- [x] I followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [x] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [x] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: www/install/steps/templates/step9.tpl [hunk 1]
```diff
@@ -16,7 +16,12 @@
                     <table cellpadding='0' cellspacing='0' border='0' class='StyleDottedHr' align='center'>
                         <tbody>
                         <tr>
-                            <td class='formValue'><input value='1' name='send_statistics' type="checkbox" checked/></td>
+                            <td class='formValue'>
+                                <div class='md-checkbox md-checkbox-inline'>
+                                    <input id='send_statistics' value='1' name='send_statistics' type='checkbox' checked='checked'/>
+                                    <label class="empty-label md-label-green" for='send_statistics'></label>
```
Comment (kduret) [inline www/install/steps/templates/step9.tpl]:
```suggestion
                                    <label class="empty-label" for='send_statistics'></label>
```

Comment (kduret) [reply www/install/steps/templates/step9.tpl]: furthermore, add this design to the same checkbox in upgrade process ;)

**Comment by adr-mo** [comment]
Hi @kduret I've pushed one commit in order to upgrade the style of the checkboxe during the upgrade process (step when asking to join the stats program)

File: www/install/steps/process/process_step8.php [hunk 2]
```diff
@@ -50,11 +51,29 @@
...
 }
 
+if (isset($parameters['widgets'])) {
+    $utilsFactory = new \CentreonLegacy\Core\Utils\Factory($dependencyInjector);
+    $utils = $utilsFactory->newUtils();
+    $widgetFactory = new \CentreonLegacy\Core\Widget\Factory($dependencyInjector, $utils);
+    foreach ($parameters['widgets'] as $widget) {
+        $installer = $widgetFactory->newInstaller($widget);
+        $id = $installer->install();
+        $install = false;
```
Comment (callapa) [inline www/install/steps/process/process_step8.php]:
```suggestion
        $install = ($id) ? true : false;
```

**Comment by adr-mo** [comment]
Thanks @callapa I've committed your suggestion ! :100:

**Pull request merged**